### PR TITLE
Fix cpu offload related slow tests

### PR DESCRIPTION
**Pull request by yiyixuxu**
This PR cleans up the cpu_offloading code for kandinsky & IF
This should fix all cpu offload tests failure related to IF and Kandisnky, e.g  https://github.com/huggingface/diffusers/actions/runs/8594679447/job/23548137528#step:9:17
https://github.com/huggingface/diffusers/actions/runs/8594679447/job/23548137612#step:9:60
https://github.com/huggingface/diffusers/actions/runs/8594679447/job/23548137335#step:9:17

 
I still have a few tests related to `from_pipe` I'm fixing here https://github.com/huggingface/diffusers/pull/7564 - but they are edge cases so don't have to worry about too much for now

### CODE REVIEW

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if.py [hunk 1]
```diff
@@ -335,9 +321,6 @@ def run_safety_checker(self, image, device, dtype):
             nsfw_detected = None
             watermark_detected = None
 
-            if hasattr(self, "unet_offload_hook") and self.unet_offload_hook is not None:
-                self.unet_offload_hook.offload()
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if.py:339]: I don't think we need to manually offload anything anymore; I don't think `unry_offload_hook` on pipeline is ever used

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img.py [hunk 2]
```diff
@@ -139,6 +138,7 @@ class IFImg2ImgPipeline(DiffusionPipeline, LoraLoaderMixin):
 
     _optional_components = ["tokenizer", "text_encoder", "safety_checker", "feature_extractor", "watermarker"]
     model_cpu_offload_seq = "text_encoder->unet"
+    _exclude_from_cpu_offload = ["watermarker"]
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img.py:141]: `whatermarker` does not work with the sequential_cpu_offload hooks, since they are very small, I'm just excluding it

**Review by DN6 — APPROVED** [review]
Nicely done 👍🏽
